src/components/common/AddressBookInput/index.tsx

```diff
@@ -13,37 +15,67 @@ const abFilterOptions = createFilterOptions({
 /**
  *  Temporary component until revamped safe components are done
  */
-const AddressBookInput = ({ name, ...props }: AddressInputProps): ReactElement => {
+const AddressBookInput = ({ name, canAdd, ...props }: AddressInputProps): ReactElement => {
   const addressBook = useAddressBook()
   const { setValue, control } = useFormContext()
   const addressValue = useWatch({ name, control })
+  const [open, setOpen] = useState(false)
 
   const addressBookEntries = Object.entries(addressBook).map(([address, name]) => ({
     label: address,
     name,
   }))
 
+  const hasVisibleOptions = useMemo(
+    () => !!addressBookEntries.filter((entry) => entry.label.includes(addressValue)).length,
+    [addressBookEntries, addressValue],
+  )
+
+  const handleOpenAutocomplete = () => {
+    setOpen((value) => !value)
+  }
+
   return (
-    <Autocomplete
-      value={addressValue || ''}
-      disabled={props.disabled}
-      readOnly={props.InputProps?.readOnly}
-      freeSolo
-      options={addressBookEntries}
-      onInputChange={(_, value) => setValue(name, value, { shouldValidate: true })}
-      filterOptions={abFilterOptions}
-      componentsProps={{
-        paper: {
-          elevation: 2,
-        },
-      }}
-      renderOption={(props, option) => (
-        <Typography component="li" variant="body2" {...props}>
-          <EthHashInfo address={option.label} name={option.name} shortAddress={false} />
+    <>
+      <Autocomplete
+        open={open}
+        onOpen={() => setOpen(true)}
+        onClose={() => setOpen(false)}
+        disableClearable
+        value={addressValue || ''}
+        disabled={props.disabled}
+        readOnly={props.InputProps?.readOnly}
+        freeSolo
+        options={addressBookEntries}
+        onInputChange={(_, value) => setValue(name, value, { shouldValidate: true })}
+        filterOptions={abFilterOptions}
+        componentsProps={{
+          paper: {
+            elevation: 2,
+          },
+        }}
+        renderOption={(props, option) => (
+          <Typography component="li" variant="body2" {...props}>
+            <EthHashInfo address={option.label} name={option.name} shortAddress={false} />
+          </Typography>
+        )}
+        renderInput={(params) => (
+          <AddressInput
+            {...params}
+            {...props}
+            name={name}
+            onOpenListClick={hasVisibleOptions ? handleOpenAutocomplete : undefined}
+            isAutocompleteOpen={open}
+          />
+        )}
+      />
+      {canAdd ? (
+        <Typography variant="body2" className={css.unknownAddress}>
+          <SvgIcon component={InfoIcon} fontSize="small" />
+          <span>You&apos;ve input an unknown address. Add it to your address book.</span>
```

Comment:
Do you think we should only show this when it's a valid address?

![image](https://github.com/safe-global/safe-wallet-web/assets/20442784/382b9394-2831-4d2d-9df8-1db9a008fc52)

I would also suggest we tweak the wording a bit and have it open the modal when clicking on "adding it to your address book".

```suggestion
          <span>This is an unknown address. Consider adding it to your address book.</span>
```